executor/src/core/graphQL-interface/GraphQL.ts

```diff
@@ -15,22 +15,27 @@ import * as PubSub from './PubSub'
 import { GraphQLScalarType } from "graphql";
 import { ad4mExecutorVersion } from '../Config';
 import * as Auth from '../agent/Auth'
-import { checkCapability } from '../agent/Auth'
+import { checkCapability, checkTokenAuthorized } from '../agent/Auth'
 import { withFilter } from 'graphql-subscriptions';
 import { OuterConfig } from '../../main';
+import path from 'path';
 
 function createResolvers(core: PerspectivismCore, config: OuterConfig) {
     const pubsub = PubSub.get()
-
+    const rootConfigPath = path.join(config.appDataPath, 'ad4m');
     return {
         Query: {
             //@ts-ignore
             agent: (parent, args, context, info) => {
+                const isAd4minCredential =  core.agentService.isAd4minCredential(context.authToken)
+                checkTokenAuthorized(rootConfigPath, context.authToken, isAd4minCredential)
```

Comment:
I would like to see the name changed (isAd4min to isAdmin), but why not put `checkTokenAuthorized()` into `AgentService` and have one line less in this block that is copied in each of these functions? (Or maybe even merge with / put into `checkCapabality()` to avoid even more copy-pasta)..

Reply:
I initially thought to put it within the `checkCapability` but then I thought maybe it's better to have it separate as it does something different than the function.

Yeah I guess it's better to be handled as a middleware in graphql.